packages/devtools_shared/lib/src/deeplink/deeplink_manager.dart

```diff
@@ -8,26 +8,84 @@ import 'dart:io';
 import 'package:meta/meta.dart';
 
 class DeeplinkManager {
-  /// A regex to retrieve the json part of the stdout.
+  /// A regex to retrieve the json part of the stdout of Android analyzer.
   ///
   /// Example stdout:
   ///
   /// Running Gradle task 'printBuildVariants'...                        10.4s
   /// ["debug","release","profile"]
-  static final _buildVariantJsonRegex = RegExp(r'(\[.*\])');
+  static final _androidBuildVariantJsonRegex = RegExp(r'(\[.*\])');
+
+  /// A regex to retrieve the json part of the stdout of iOS analyzer.
+  ///
+  /// Example stdout:
+  ///
+  /// {"configurations":["Debug","Release","Profile"],"targets":["Runner","RunnerTests"]}
+  static final _iosBuildOptionsJsonRegex = RegExp(r'({.*})');
+
   static const kErrorField = 'error';
   static const kOutputJsonField = 'json';
 
+  static final _outputFilePathRegex = RegExp(r'result saved in (.*.json)');
+
   @visibleForTesting
-  Future<ProcessResult> runProcess(String executable, List<String> arguments) {
-    return Process.run(executable, arguments);
+  Future<ProcessResult> runProcess(
+    String executable, {
+    required List<String> arguments,
+    required String workingDirectory,
+  }) {
+    return Process.run(
+      executable,
+      arguments,
+      workingDirectory: workingDirectory,
+    );
   }
 
-  Future<Map<String, Object?>> getBuildVariants({
+  Future<ProcessResult> _runFlutterCommand(List<String> arguments) {
+    final String flutterRoot = Platform.environment['FLUTTER_ROOT']!;
+    return runProcess(
+      'flutter',
+      arguments: arguments,
+      workingDirectory: flutterRoot,
+    );
```

Comment:
`FLUTTER_ROOT` will be the SDK folder (without `bin`), so you probably need to combine these.

I'm also not sure that `workingDirectory` is always searched for the executable, so I think it should be prepended to the `executable`. So you might need something like:

```dart
    return runProcess(
      path.join(flutterRoot, 'bin', Platform.isWindows ? 'flutter.bat' : 'flutter'),
      arguments: arguments,
    );
```

I would also suggest not using `!` to assume this will always be set, in case DevTools is running in a context other than VS Code, or a version of the Dart plugin that doesn't include https://github.com/Dart-Code/Dart-Code/issues/4722 (such as the current stable version).

You should be able to test this using the pre-release extension, but let me know if that doesn't seem to be the case.

<img width="703" alt="Dart pre-release versions" src="https://user-images.githubusercontent.com/1078012/218796750-90a70b0b-4957-468c-8eb4-688095cde112.png">

Reply:
I added the logic to get the proper path and also handler when env var is not set